models/_generate_layers.py

```diff
@@ -496,6 +497,38 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer):
+  """
+  Inspects the given layer, extracts unwanted parameters and then returns a string that is ready to be used in the super
+  call.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  super_call = lines[0]
+  call = super_call.split(".")[1]
+  call_pruned = call[9:-1]
+
+  # remove excluded params
+  split = call_pruned.split(",")
+  tup_ls = list(map(lambda x: x.split("="), split))
+  for idx, param in enumerate(tup_ls):
+    if param[0].strip() in LayerSignature.IgnoreParamNames or param[0].strip() == "batch_norm":
+      tup_ls.pop(idx)
+  tup_ls = list(map(lambda x: "=".join(x), tup_ls))
+  tup_ls = list(map(lambda x: x.strip(), tup_ls))
+  params = ", ".join(tup_ls)
+  return params
```

Comment:
I don't exactly understand the return. It would return e.g. sth like `"key='value'"`.
This would be used in your code above like `super().__init__({key='value'})`.
But this would be a syntax error. You either would want `super().__init__(dict(key='value'))` or `super().__init__({'key': 'value'})`.

But the return value is anyway not really clear (or clearly documented).

Reply:
I think you are missunderstanding something here (which most likely was due to the code being unclear). 
The function returns a pure string, not a dict of some kind, which the directly can be used in the write function. 
I hope this is more clear now.

Reply:
This was already clear. Maybe you misunderstood what I wrote? It would lead to the code I wrote in my comment (because of `print(f" super().__init__({params})", file=f)` above).

Reply:
Ah sorry, I read the code wrongly. The `{`,`}` will not be printed but are used for the f-string. So it would not produce `super().__init__({key='value'})` but `super().__init__(key='value')`. So this looks fine then.

Reply:
Most likely I am missinderstanding something 😅 The curly brackets here are just for the format string, I am not sure why that would cause problems.